pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdaterTest.java

```diff
@@ -53,49 +53,57 @@ public class FlushThresholdUpdaterTest {
   public void testFlushThresholdUpdateManager() {
     FlushThresholdUpdateManager flushThresholdUpdateManager = new FlushThresholdUpdateManager();
 
-    // Flush threshold rows larger than 0 - DefaultFlushThresholdUpdater should be returned
-    FlushThresholdUpdater defaultFlushThresholdUpdater = flushThresholdUpdateManager
-        .getFlushThresholdUpdater(mockStreamConfig(StreamConfig.DEFAULT_FLUSH_THRESHOLD_ROWS));
-    assertTrue(defaultFlushThresholdUpdater instanceof DefaultFlushThresholdUpdater);
-    assertEquals(((DefaultFlushThresholdUpdater) defaultFlushThresholdUpdater).getTableFlushSize(),
+    // Neither flush threshold rows nor segment size is set - DefaultFlushThresholdUpdater should be returned
+    FlushThresholdUpdater flushThresholdUpdater =
+        flushThresholdUpdateManager.getFlushThresholdUpdater(mockStreamConfig(-1, -1));
+    assertTrue(flushThresholdUpdater instanceof DefaultFlushThresholdUpdater);
+    assertEquals(((DefaultFlushThresholdUpdater) flushThresholdUpdater).getTableFlushSize(),
         StreamConfig.DEFAULT_FLUSH_THRESHOLD_ROWS);
 
+    // Flush threshold rows larger than 0 - DefaultFlushThresholdUpdater should be returned
```

Comment:
Just to make sure (I see that this is already the case as we compute the row based config first): 

When both row and segment size based configurations are configured, the original code picked the row based option?

This is important as changing this will make the code not backward compatible.

Reply:
For existing table configs, we follow this order; we don't allow such config in new added/updated table config as we validate only one can be configured